### ISSUE TITLE
Failed to mount folders in Linux guest. This is usually because the "vboxsf" file system is not available.

### ISSUE THREAD

**Issue by vaynard** [puphpet/puphpet#2144]
HOST: Windows 8.1 Pro X64

```
V:\dmp-sf-vagrant>vagrant up
Bringing machine 'local' up with 'virtualbox' provider...
==> local: Importing base box 'puphpet/ubuntu1404-x32'...
==> local: Matching MAC address for NAT networking...
==> local: Checking if box 'puphpet/ubuntu1404-x32' is up to date...
==> local: Setting the name of the VM: dmp-sf-vagrant_local_1454426620009_34552
==> local: Preparing SMB shared folders...
==> local: Clearing any previously set network interfaces...
==> local: Preparing network interfaces based on configuration...
    local: Adapter 1: nat
    local: Adapter 2: hostonly
==> local: Forwarding ports...
    local: 22 (guest) => 8022 (host) (adapter 1)
    local: 22 (guest) => 2222 (host) (adapter 1)
==> local: Running 'pre-boot' VM customizations...
==> local: Booting VM...
==> local: Waiting for machine to boot. This may take a few minutes...
    local: SSH address: 127.0.0.1:2222
    local: SSH username: vagrant
    local: SSH auth method: private key
==> local: Machine booted and ready!
GuestAdditions versions on your host (5.0.14) and guest (5.0.10) do not match.
Reading package lists...
Building dependency tree...
dkms is already the newest version.
linux-headers-3.19.0-25-generic is already the newest version.
0 upgraded, 0 newly installed, 0 to remove and 3 not upgraded.
Copy iso file C:\Program Files\Oracle\VirtualBox\VBoxGuestAdditions.iso into the
 box /tmp/VBoxGuestAdditions.iso
mount: block device /tmp/VBoxGuestAdditions.iso is write-protected, mounting rea
d-only
Installing Virtualbox Guest Additions 5.0.14 - guest version is 5.0.10
Verifying archive integrity... All good.
Uncompressing VirtualBox 5.0.14 Guest Additions for Linux............
VirtualBox Guest Additions installer
Removing installed version 5.0.10 of VirtualBox Guest Additions...
Stopping VirtualBox Additions ...fail!
(Cannot unload module vboxguest)
Removing existing VirtualBox DKMS kernel modules ...done.
Removing existing VirtualBox non-DKMS kernel modules ...done.
Stopping VirtualBox Guest Addition service  ...done.
Copying additional installer modules ...
Installing additional modules ...
Removing existing VirtualBox DKMS kernel modules ...done.
Removing existing VirtualBox non-DKMS kernel modules ...done.
Building the VirtualBox Guest Additions kernel modules ...done.
Doing non-kernel setup of the Guest Additions ...done.
You should restart your guest to make sure the new modules are actually used

Installing the Window System drivers
Could not find the X.Org or XFree86 Window System, skipping.
An error occurred during installation of VirtualBox Guest Additions 5.0.14. Some
 functionality may not work as intended.
In most cases it is OK that the "Window System drivers" installation failed.
==> local: Checking for guest additions in VM...
==> local: Setting hostname...
==> local: Configuring and enabling network interfaces...
==> local: Mounting SMB shared folders...
    local: Installing SMB "mount.cifs"...
    local: V:/dmp-sf-vagrant/dmp-sf => /var/www
Failed to mount folders in Linux guest. This is usually because
the "vboxsf" file system is not available. Please verify that
the guest additions are properly installed in the guest and
can work properly. The command attempted was:

PASSWORDHIDDENmPASSWORDHIDDENoPASSWORDHIDDENuPASSWORDHIDDENnPASSWORDHIDDENtPASSW
ORDHIDDEN PASSWORDHIDDEN-PASSWORDHIDDENtPASSWORDHIDDEN PASSWORDHIDDENcPASSWORDHI
DDENiPASSWORDHIDDENfPASSWORDHIDDENsPASSWORDHIDDEN PASSWORDHIDDEN-PASSWORDHIDDENo
PASSWORDHIDDEN PASSWORDHIDDENuPASSWORDHIDDENiPASSWORDHIDDENdPASSWORDHIDDEN=PASSW
ORDHIDDEN`PASSWORDHIDDENiPASSWORDHIDDENdPASSWORDHIDDEN PASSWORDHIDDEN-PASSWORDHI
DDENuPASSWORDHIDDEN PASSWORDHIDDENwPASSWORDHIDDENwPASSWORDHIDDENwPASSWORDHIDDEN-
PASSWORDHIDDENdPASSWORDHIDDENaPASSWORDHIDDENtPASSWORDHIDDENaPASSWORDHIDDEN`PASSW
ORDHIDDEN,PASSWORDHIDDENgPASSWORDHIDDENiPASSWORDHIDDENdPASSWORDHIDDEN=PASSWORDHI
DDEN`PASSWORDHIDDENgPASSWORDHIDDENePASSWORDHIDDENtPASSWORDHIDDENePASSWORDHIDDENn
PASSWORDHIDDENtPASSWORDHIDDEN PASSWORDHIDDENgPASSWORDHIDDENrPASSWORDHIDDENoPASSW
ORDHIDDENuPASSWORDHIDDENpPASSWORDHIDDEN PASSWORDHIDDENwPASSWORDHIDDENwPASSWORDHI
DDENwPASSWORDHIDDEN-PASSWORDHIDDENdPASSWORDHIDDENaPASSWORDHIDDENtPASSWORDHIDDENa
PASSWORDHIDDEN PASSWORDHIDDEN|PASSWORDHIDDEN PASSWORDHIDDENcPASSWORDHIDDENuPASSW
ORDHIDDENtPASSWORDHIDDEN PASSWORDHIDDEN-PASSWORDHIDDENdPASSWORDHIDDEN:PASSWORDHI
DDEN PASSWORDHIDDEN-PASSWORDHIDDENfPASSWORDHIDDEN3PASSWORDHIDDEN`PASSWORDHIDDEN,
PASSWORDHIDDENdPASSWORDHIDDENiPASSWORDHIDDENrPASSWORDHIDDEN_PASSWORDHIDDENmPASSW
ORDHIDDENoPASSWORDHIDDENdPASSWORDHIDDENePASSWORDHIDDEN=PASSWORDHIDDEN0PASSWORDHI
DDEN7PASSWORDHIDDEN7PASSWORDHIDDEN5PASSWORDHIDDEN,PASSWORDHIDDENfPASSWORDHIDDENi
PASSWORDHIDDENlPASSWORDHIDDENePASSWORDHIDDEN_PASSWORDHIDDENmPASSWORDHIDDENoPASSW
ORDHIDDENdPASSWORDHIDDENePASSWORDHIDDEN=PASSWORDHIDDEN0PASSWORDHIDDEN6PASSWORDHI
DDEN6PASSWORDHIDDEN4PASSWORDHIDDEN,PASSWORDHIDDENsPASSWORDHIDDENePASSWORDHIDDENc
PASSWORDHIDDEN=PASSWORDHIDDENnPASSWORDHIDDENtPASSWORDHIDDENlPASSWORDHIDDENmPASSW
ORDHIDDEN,PASSWORDHIDDENcPASSWORDHIDDENrPASSWORDHIDDENePASSWORDHIDDENdPASSWORDHI
DDENePASSWORDHIDDENnPASSWORDHIDDENtPASSWORDHIDDENiPASSWORDHIDDENaPASSWORDHIDDENl
PASSWORDHIDDENsPASSWORDHIDDEN=PASSWORDHIDDEN/PASSWORDHIDDENePASSWORDHIDDENtPASSW
ORDHIDDENcPASSWORDHIDDEN/PASSWORDHIDDENsPASSWORDHIDDENmPASSWORDHIDDENbPASSWORDHI
DDEN_PASSWORDHIDDENcPASSWORDHIDDENrPASSWORDHIDDENePASSWORDHIDDENdPASSWORDHIDDENs
PASSWORDHIDDEN_PASSWORDHIDDENcPASSWORDHIDDENaPASSWORDHIDDEN0PASSWORDHIDDENbPASSW
ORDHIDDEN6PASSWORDHIDDEN9PASSWORDHIDDENcPASSWORDHIDDENaPASSWORDHIDDEN3PASSWORDHI
DDEN7PASSWORDHIDDENbPASSWORDHIDDEN6PASSWORDHIDDEN7PASSWORDHIDDEN7PASSWORDHIDDEN1
PASSWORDHIDDENePASSWORDHIDDEN8PASSWORDHIDDEN0PASSWORDHIDDEN3PASSWORDHIDDENdPASSW
ORDHIDDEN9PASSWORDHIDDEN6PASSWORDHIDDEN7PASSWORDHIDDEN1PASSWORDHIDDENcPASSWORDHI
DDEN6PASSWORDHIDDEN2PASSWORDHIDDEN2PASSWORDHIDDEN6PASSWORDHIDDENdPASSWORDHIDDENd
PASSWORDHIDDEN4PASSWORDHIDDEN PASSWORDHIDDEN/PASSWORDHIDDEN/PASSWORDHIDDEN/PASSW
ORDHIDDENcPASSWORDHIDDENaPASSWORDHIDDEN0PASSWORDHIDDENbPASSWORDHIDDEN6PASSWORDHI
DDEN9PASSWORDHIDDENcPASSWORDHIDDENaPASSWORDHIDDEN3PASSWORDHIDDEN7PASSWORDHIDDENb
PASSWORDHIDDEN6PASSWORDHIDDEN7PASSWORDHIDDEN7PASSWORDHIDDEN1PASSWORDHIDDENePASSW
ORDHIDDEN8PASSWORDHIDDEN0PASSWORDHIDDEN3PASSWORDHIDDENdPASSWORDHIDDEN9PASSWORDHI
DDEN6PASSWORDHIDDEN7PASSWORDHIDDEN1PASSWORDHIDDENcPASSWORDHIDDEN6PASSWORDHIDDEN2
PASSWORDHIDDEN2PASSWORDHIDDEN6PASSWORDHIDDENdPASSWORDHIDDENdPASSWORDHIDDEN4PASSW
ORDHIDDEN PASSWORDHIDDEN/PASSWORDHIDDENvPASSWORDHIDDENaPASSWORDHIDDENrPASSWORDHI
DDEN/PASSWORDHIDDENwPASSWORDHIDDENwPASSWORDHIDDENwPASSWORDHIDDEN
PASSWORDHIDDENmPASSWORDHIDDENoPASSWORDHIDDENuPASSWORDHIDDENnPASSWORDHIDDENtPASSW
ORDHIDDEN PASSWORDHIDDEN-PASSWORDHIDDENtPASSWORDHIDDEN PASSWORDHIDDENcPASSWORDHI
DDENiPASSWORDHIDDENfPASSWORDHIDDENsPASSWORDHIDDEN PASSWORDHIDDEN-PASSWORDHIDDENo
PASSWORDHIDDEN PASSWORDHIDDENuPASSWORDHIDDENiPASSWORDHIDDENdPASSWORDHIDDEN=PASSW
ORDHIDDEN`PASSWORDHIDDENiPASSWORDHIDDENdPASSWORDHIDDEN PASSWORDHIDDEN-PASSWORDHI
DDENuPASSWORDHIDDEN PASSWORDHIDDENwPASSWORDHIDDENwPASSWORDHIDDENwPASSWORDHIDDEN-
PASSWORDHIDDENdPASSWORDHIDDENaPASSWORDHIDDENtPASSWORDHIDDENaPASSWORDHIDDEN`PASSW
ORDHIDDEN,PASSWORDHIDDENgPASSWORDHIDDENiPASSWORDHIDDENdPASSWORDHIDDEN=PASSWORDHI
DDEN`PASSWORDHIDDENiPASSWORDHIDDENdPASSWORDHIDDEN PASSWORDHIDDEN-PASSWORDHIDDENg
PASSWORDHIDDEN PASSWORDHIDDENwPASSWORDHIDDENwPASSWORDHIDDENwPASSWORDHIDDEN-PASSW
ORDHIDDENdPASSWORDHIDDENaPASSWORDHIDDENtPASSWORDHIDDENaPASSWORDHIDDEN`PASSWORDHI
DDEN,PASSWORDHIDDENdPASSWORDHIDDENiPASSWORDHIDDENrPASSWORDHIDDEN_PASSWORDHIDDENm
PASSWORDHIDDENoPASSWORDHIDDENdPASSWORDHIDDENePASSWORDHIDDEN=PASSWORDHIDDEN0PASSW
ORDHIDDEN7PASSWORDHIDDEN7PASSWORDHIDDEN5PASSWORDHIDDEN,PASSWORDHIDDENfPASSWORDHI
DDENiPASSWORDHIDDENlPASSWORDHIDDENePASSWORDHIDDEN_PASSWORDHIDDENmPASSWORDHIDDENo
PASSWORDHIDDENdPASSWORDHIDDENePASSWORDHIDDEN=PASSWORDHIDDEN0PASSWORDHIDDEN6PASSW
ORDHIDDEN6PASSWORDHIDDEN4PASSWORDHIDDEN,PASSWORDHIDDENsPASSWORDHIDDENePASSWORDHI
DDENcPASSWORDHIDDEN=PASSWORDHIDDENnPASSWORDHIDDENtPASSWORDHIDDENlPASSWORDHIDDENm
PASSWORDHIDDEN,PASSWORDHIDDENcPASSWORDHIDDENrPASSWORDHIDDENePASSWORDHIDDENdPASSW
ORDHIDDENePASSWORDHIDDENnPASSWORDHIDDENtPASSWORDHIDDENiPASSWORDHIDDENaPASSWORDHI
DDENlPASSWORDHIDDENsPASSWORDHIDDEN=PASSWORDHIDDEN/PASSWORDHIDDENePASSWORDHIDDENt
PASSWORDHIDDENcPASSWORDHIDDEN/PASSWORDHIDDENsPASSWORDHIDDENmPASSWORDHIDDENbPASSW
ORDHIDDEN_PASSWORDHIDDENcPASSWORDHIDDENrPASSWORDHIDDENePASSWORDHIDDENdPASSWORDHI
DDENsPASSWORDHIDDEN_PASSWORDHIDDENcPASSWORDHIDDENaPASSWORDHIDDEN0PASSWORDHIDDENb
PASSWORDHIDDEN6PASSWORDHIDDEN9PASSWORDHIDDENcPASSWORDHIDDENaPASSWORDHIDDEN3PASSW
ORDHIDDEN7PASSWORDHIDDENbPASSWORDHIDDEN6PASSWORDHIDDEN7PASSWORDHIDDEN7PASSWORDHI
DDEN1PASSWORDHIDDENePASSWORDHIDDEN8PASSWORDHIDDEN0PASSWORDHIDDEN3PASSWORDHIDDENd
PASSWORDHIDDEN9PASSWORDHIDDEN6PASSWORDHIDDEN7PASSWORDHIDDEN1PASSWORDHIDDENcPASSW
ORDHIDDEN6PASSWORDHIDDEN2PASSWORDHIDDEN2PASSWORDHIDDEN6PASSWORDHIDDENdPASSWORDHI
DDENdPASSWORDHIDDEN4PASSWORDHIDDEN PASSWORDHIDDEN/PASSWORDHIDDEN/PASSWORDHIDDEN/
PASSWORDHIDDENcPASSWORDHIDDENaPASSWORDHIDDEN0PASSWORDHIDDENbPASSWORDHIDDEN6PASSW
ORDHIDDEN9PASSWORDHIDDENcPASSWORDHIDDENaPASSWORDHIDDEN3PASSWORDHIDDEN7PASSWORDHI
DDENbPASSWORDHIDDEN6PASSWORDHIDDEN7PASSWORDHIDDEN7PASSWORDHIDDEN1PASSWORDHIDDENe
PASSWORDHIDDEN8PASSWORDHIDDEN0PASSWORDHIDDEN3PASSWORDHIDDENdPASSWORDHIDDEN9PASSW
ORDHIDDEN6PASSWORDHIDDEN7PASSWORDHIDDEN1PASSWORDHIDDENcPASSWORDHIDDEN6PASSWORDHI
DDEN2PASSWORDHIDDEN2PASSWORDHIDDEN6PASSWORDHIDDENdPASSWORDHIDDENdPASSWORDHIDDEN4
PASSWORDHIDDEN PASSWORDHIDDEN/PASSWORDHIDDENvPASSWORDHIDDENaPASSWORDHIDDENrPASSW
ORDHIDDEN/PASSWORDHIDDENwPASSWORDHIDDENwPASSWORDHIDDENwPASSWORDHIDDEN

The error output from the last command was:

mount.cifs: bad UNC (///ca0b69ca37b6771e803d9671c6226dd4)
```

**Comment by MariusRumpf** [comment]
Similar on OSX 10.11.3
Vagrant 1.8.1
```
GuestAdditions versions on your host (5.0.14) and guest (5.0.10) do not match.
Reading package lists...
Building dependency tree...
dkms is already the newest version.
linux-headers-3.19.0-25-generic is already the newest version.
0 upgraded, 0 newly installed, 0 to remove and 3 not upgraded.
Copy iso file /Applications/VirtualBox.app/Contents/MacOS/VBoxGuestAdditions.iso into the box /tmp/VBoxGuestAdditions.iso
mount: block device /tmp/VBoxGuestAdditions.iso is write-protected, mounting read-only
Installing Virtualbox Guest Additions 5.0.14 - guest version is 5.0.10
Verifying archive integrity... All good.
Uncompressing VirtualBox 5.0.14 Guest Additions for Linux............
VirtualBox Guest Additions installer
Removing installed version 5.0.10 of VirtualBox Guest Additions...
Stopping VirtualBox Additions ...fail!
(Cannot unload module vboxguest)
Removing existing VirtualBox DKMS kernel modules ...done.
Removing existing VirtualBox non-DKMS kernel modules ...done.
Stopping VirtualBox Guest Addition service  ...done.
Copying additional installer modules ...
Installing additional modules ...
Removing existing VirtualBox DKMS kernel modules ...done.
Removing existing VirtualBox non-DKMS kernel modules ...done.
Building the VirtualBox Guest Additions kernel modules ...done.
Doing non-kernel setup of the Guest Additions ...done.
You should restart your guest to make sure the new modules are actually used

Installing the Window System drivers
Could not find the X.Org or XFree86 Window System, skipping.
An error occurred during installation of VirtualBox Guest Additions 5.0.14. Some functionality may not work as intended.
In most cases it is OK that the "Window System drivers" installation failed.
...
```

**Comment by dyron** [comment]
Same on Windows 10 and Vagrant 1.8.1.
This comes from using sync_type as 'smb'.
'default' seems to work.

**Comment by ReFury** [comment]
Changing sync_type to default or nfs doesn't help. Same issue.

**Comment by zablik** [comment]
Same issue. 
Win 10
Vagrant 1.8.1
VirtualBox 5.0.16

**Comment by vielhuber** [comment]
Same here.
Win 10
Vagrant 1.8.1
VirtualBox 5.0.16

**Comment by d-ph** [comment]
Hey guys,

tl;dr:

Replace this:
```
smb_host: ''
smb_username: ''
smb_password: ''
```
with this:
```
smb_host: ~
smb_username: ~
smb_password: ~
```
In `config.yaml`

---

`smb_host`, `smb_username` and `smb_password` must not be empty strings in `config.yaml`.

**Comment by alejoluc** [comment]
The solution proposed by @d-ph  fixed it for me. Please merge into main branch

**Comment by jtreminio** [comment]
Sorry for the ridiculous lag in getting this done: https://github.com/puphpet/puphpet/commit/c6813331dbf69db8d0da71f3025a8d1f6d79358c
